confgenerator/self_logs.go

```diff
@@ -108,6 +108,16 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "fluent-bit-self-log-regex-parsing")...)
+	if logLevel == "debug" {
+		out = append(out, fluentbit.Component{
+			Kind: "FILTER",
+			Config: map[string]string{
+				"Name":    "grep",
+				"Match":   fluentBitSelfLogsTag,
+				"Exclude": "severity DEBUG",
```

Comment:
I believe this exclusion filter won't work since the `severity field` at this point hasn't been transformed correctly to a logging api friendly format (this func comment is misleading, see other comment). The translation happens in `generateSelfLogsProcessingComponents()` to make the translation for all self logs once.

Some options for this to work  : 
- Move filter to `generateSelfLogsProcessingComponents()` after severity field transform. The exclude filter would look like (may fail due to backslashes if this hasn't been fixed in the current fluent-bit version in the ops agent) :
```
"Exclude": "logging.googleapis.com/severity DEBUG",
```
- Keep filter here in `generateFluentBitSelfLogsComponents()` and use lowercase `debug` : 
```
"Exclude": "severity debug",
```